File: base/--build/tmp/kapt3/stubs/debug/com/shopify/shopifyapp/ordersection/adapters/OrderListAdapter.java
package com.baytrendy.shopifyapp.ordersection.adapters;

import java.lang.System;

@kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000>\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010!\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\b\n\u0000\n\u0002\u0010\u0002\n\u0002\b\u0004\n\u0002\u0018\u0002\n\u0002\b\u0002\u0018\u00002\b\u0012\u0004\u0012\u00020\u00020\u0001B\u0007\b\u0007\u00a2\u0006\u0002\u0010\u0003J\b\u0010\u000f\u001a\u00020\u0010H\u0016J\u0018\u0010\u0011\u001a\u00020\u00122\u0006\u0010\u0013\u001a\u00020\u00022\u0006\u0010\u0014\u001a\u00020\u0010H\u0016J\u0018\u0010\u0015\u001a\u00020\u00022\u0006\u0010\u0016\u001a\u00020\u00172\u0006\u0010\u0018\u001a\u00020\u0010H\u0016J \u0010\t\u001a\u00020\u00122\u000e\u0010\u0004\u001a\n\u0012\u0004\u0012\u00020\u0006\u0018\u00010\u00052\b\u0010\r\u001a\u0004\u0018\u00010\u000eR\"\u0010\u0004\u001a\n\u0012\u0004\u0012\u00020\u0006\u0018\u00010\u0005X\u0086\u000e\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\u0007\u0010\b\"\u0004\b\t\u0010\nR\u0010\u0010\u000b\u001a\u0004\u0018\u00010\fX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\r\u001a\u0004\u0018\u00010\u000eX\u0082\u000e\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u0019"}, d2 = {"Lcom/shopify/shopifyapp/ordersection/adapters/OrderListAdapter;", "Landroidx/recyclerview/widget/RecyclerView$Adapter;", "Lcom/shopify/shopifyapp/ordersection/viewholders/OrderItem;", "()V", "data", "", "Lcom/shopify/buy3/Storefront$OrderEdge;", "getData", "()Ljava/util/List;", "setData", "(Ljava/util/List;)V", "layoutInflater", "Landroid/view/LayoutInflater;", "model", "Lcom/shopify/shopifyapp/ordersection/viewmodels/OrderListViewModel;", "getItemCount", "", "onBindViewHolder", "", "holder", "position", "onCreateViewHolder", "parent", "Landroid/view/ViewGroup;", "viewType", "base_debug"})
public final class OrderListAdapter extends androidx.recyclerview.widget.RecyclerView.Adapter<com.baytrendy.shopifyapp.ordersection.viewholders.OrderItem> {
    @org.jetbrains.annotations.Nullable()
    private java.util.List<com.shopify.buy3.Storefront.OrderEdge> data;
    private android.view.LayoutInflater layoutInflater;
    private com.baytrendy.shopifyapp.ordersection.viewmodels.OrderListViewModel model;
    
    @org.jetbrains.annotations.Nullable()
    public final java.util.List<com.shopify.buy3.Storefront.OrderEdge> getData() {
        return null;
    }
    
    public final void setData(@org.jetbrains.annotations.Nullable()
    java.util.List<com.shopify.buy3.Storefront.OrderEdge> p0) {
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public com.baytrendy.shopifyapp.ordersection.viewholders.OrderItem onCreateViewHolder(@org.jetbrains.annotations.NotNull()
    android.view.ViewGroup parent, int viewType) {
        return null;
    }
    
    @java.lang.Override()
    public void onBindViewHolder(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.ordersection.viewholders.OrderItem holder, int position) {
    }
    
    @java.lang.Override()
    public int getItemCount() {
        return 0;
    }
    
    public final void setData(@org.jetbrains.annotations.Nullable()
    java.util.List<com.shopify.buy3.Storefront.OrderEdge> data, @org.jetbrains.annotations.Nullable()
    com.baytrendy.shopifyapp.ordersection.viewmodels.OrderListViewModel model) {
    }
    
    @javax.inject.Inject()
    public OrderListAdapter() {
        super();
    }
}

File: base/src/main/java/com/baytrendy/shopifyapp/ordersection/adapters/OrderListAdapter.kt
package com.baytrendy.shopifyapp.ordersection.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView

import com.shopify.buy3.Storefront
import com.baytrendy.shopifyapp.R
import com.baytrendy.shopifyapp.databinding.MOrderitemBinding
import com.baytrendy.shopifyapp.ordersection.models.Order
import com.baytrendy.shopifyapp.ordersection.viewholders.OrderItem
import com.baytrendy.shopifyapp.ordersection.viewmodels.OrderListViewModel
import com.baytrendy.shopifyapp.utils.CurrencyFormatter

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

import javax.inject.Inject

class OrderListAdapter @Inject
constructor() : RecyclerView.Adapter<OrderItem>() {
    var data: MutableList<Storefront.OrderEdge>? = null
    private var layoutInflater: LayoutInflater? = null
    private var model: OrderListViewModel? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OrderItem {
        val binding = DataBindingUtil.inflate<MOrderitemBinding>(LayoutInflater.from(parent.context), R.layout.m_orderitem, parent, false)
        binding.orderdetails.textSize = 11f
        binding.orderdetails.setTextColor(binding.root.context.resources.getColor(R.color.colorPrimaryDark))
        binding.orderno.textSize = 11f
        binding.name.textSize = 11f
        binding.placedontext.textSize = 11f
        binding.date.textSize = 11f
        binding.totalspendingtext.textSize = 11f
        binding.ordernoheading.textSize = 11f
        binding.boughtforheading.textSize = 11f
        binding.boughtfor.textSize = 11f
        binding.ordernoheading.textSize = 11f
        binding.totalspending.textSize = 11f
        return OrderItem(binding)
    }

    override fun onBindViewHolder(holder: OrderItem, position: Int) {
        try {
            val order = Order()
            order.orderEdge = data?.get(position)!!.node
            order.ordernumber = data?.get(position)!!.node.orderNumber!!.toString()
            order.name = data?.get(position)!!.node.name
            val sdf2 = SimpleDateFormat("MMM dd yyyy", Locale.getDefault())
            val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.getDefault())
            val expiretime = sdf.parse(data?.get(position)!!.node.processedAt.toLocalDateTime().toString())
            val time = sdf2.format(expiretime!!)
            order.date = time
            order.price = CurrencyFormatter.setsymbol(data?.get(position)!!.node.totalPriceV2.amount, data?.get(position)!!.node.totalPriceV2.currencyCode.toString())
            order.status = data?.get(position)!!.node.statusUrl
            if (data?.get(position)!!.node.shippingAddress != null) {
                holder.binding.boughtfor.visibility = View.VISIBLE
                holder.binding.boughtforheading.visibility = View.VISIBLE
                order.boughtfor = data?.get(position)!!.node.shippingAddress.firstName + " " + data?.get(position)!!.node.shippingAddress.lastName
            } else {
                holder.binding.boughtfor.visibility = View.GONE
                holder.binding.boughtforheading.visibility = View.GONE
            }
            holder.binding.order = order
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun getItemCount(): Int {
        return data!!.size
    }

    fun setData(data: MutableList<Storefront.OrderEdge>?, model: OrderListViewModel?) {
        this.data = data
        this.model = model
    }
}
